File: Trilha Básica/Conceitos básicos de POO - Exercício Cozinha/Ingrediente.java
import java.time.LocalDate;

public class Ingrediente{

   private String nome;
   private LocalDate dataValidade;

   public Ingrediente(String nome, LocalDate dataValidade){
        this.nome = nome;
        this.dataValidade = dataValidade;
   }

   public void setNome(String nome){
        this.nome = nome;
   }
   public String getNome(){
        return this.nome;
   }
   public void setDataValidade(LocalDate dataValidade){
        this.dataValidade = dataValidade;
   }
   public LocalDate getDataValidade(){
        return this.dataValidade;
   }

   public String toString(){
        return "Ingrediente: " + this.nome + " - Validade: " + this.dataValidade; 
   }
}

File: Trilha PHP/Atividade POO/Ingrediente.php
<?php

class Ingrediente
{
    private string $nome;
    private DateTime $dataValidade;
    
    /**
     * @return mixed
     */
    public function getNome()
    {
        return $this->nome;
    }

    /**
     * @return mixed
     */
    public function getDataValidade()
    {
        return $this->dataValidade;
    }

    /**
     * @param mixed $nome
     */
    public function setNome($nome)
    {
        $this->nome = $nome;
    }

    /**
     * @param mixed $dataValidade
     */
    public function setDataValidade($dataValidade)
    {
        $this->dataValidade = $dataValidade;
    }
    
    public function mostrarIngredientes(){
        return "\nIngrediente: " . $this->getNome() . 
        "\nData de Validade: " . $this->getDataValidade()->format('d/m/Y') .
        "</br>"; 
       
    }
        
}

?>

